vertx-web/src/test/java/io/vertx/ext/web/handler/sockjs/SockJSHandlerTest.java

```diff
@@ -380,8 +380,7 @@ public void testWebContext() throws Exception {
         Session oldSession = sock.webSession();
         Session session = handler.newSession(sock.routingContext());
         User user = User.create(principal);
-        sock.routingContext().setUser(user);
-        handler.flush(sock.routingContext()).result();
+        handler.setUser(sock.routingContext(), user);
```

Comment:
The timing for the test will be correct if we don't discard the callback/future.